python/ray/_private/utils.py

```diff
@@ -564,7 +564,15 @@ def get_num_cpus(
             "To correctly detect CPUs, unset the env var: "
             "`RAY_USE_MULTIPROCESSING_CPU_COUNT`."
         )
-        return cpu_count
+        return multiprocessing.cpu_count()
+
+    if hasattr(os, "sched_getaffinity"):
+        # Reflects the real CPU count available to the calling thread of the process.
```

Comment:
This could use a bit more explanation -- why is CPU count available to the calling thread of this process important?
What's important is the CPUs available to the Ray worker process of the Ray node that this process is starting.
Are the CPUs available to the Ray worker processes on this Ray node necessarily the same as the CPUs available to the `ray start` process?

Also, could you explain more, in documentation/comments, about the type of use-case this is useful for?